### ISSUE TITLE
Support lightning network payments

### ISSUE THREAD

**Issue by philippgille** [mycelium-com/wallet-android#419]
The Lightning network is in development and will probably be the de facto standard for Bitcoin microtransactions in the near future. Not many wallets support lightning yet though. Are you planning to support it?

Info about Lightning:
- https://lightning.engineering/
- http://dev.lightning.community/

**Comment by Giszmo** [comment]
I couldn't agree more with your assessment but I think once it takes off, it will be on every wallet's radar and until then it will not get the necessary priority.

**Comment by jorijnsmit** [comment]
Sounds a bit like the chicken egg dilemma no? I think Mycelium supporting Lightning will help the protocol taking off :sunglasses:

**Comment by nilskp** [comment]
Depends on #379.

**Comment by Giszmo** [comment]
@nilskp no, lightning network does not depend on #379. This is exactly why we push modularization over individual features. With modularization, lightning can be a separate module, integrated in the wallet, with the module using segwit but the main wallet still not using segwit.

We might also provide a trezor module that has segwit, so trezor users can use mycelium for their hardware wallets again. This would neither require the main wallet to use SegWit.

**Comment by nilskp** [comment]
@Giszmo Not quite sure what you mean. As you write

> lightning can be a separate module, integrated in the wallet, with the module using segwit

The only way I can interpret that, is that #379 is specifically not a module. Is that the case? If so, is there another ticket for implementing SegWit as a module instead?

Or do you mean something else entirely?

**Comment by Giszmo** [comment]
Bitcoin is not a module yet. We might turn bitcoin into a module (at some point that was the plan) but as this module would be an internal module, it might be much work for no user feature. For now we have to get modularized mbw rolled out to all users, get feedback, get modularization a bit more polished, so that third party devs can provide their modules without guidance from mbw staff and then we can focus fully on other issues.

**Comment by nilskp** [comment]
@Giszmo It's still not clear to me why this ticket does not depend on SegWit support, as defined in ticket #379.

**Comment by TheBlueMatt** [comment]
[shill]
So lightning implementation tends to be a ton of work - not only getting the protocol implemented but also handling a million edge cases that rarely crop up in production correctly (to ensure no attacker can otherwise steal your coins) and also automating channel actions to ensure payments work reliably when users need them is nontrivial.
I've been working on https://github.com/rust-bitcoin/rust-lightning for about a year now to provide a lightning *library* instead of standalone node, ie designed to be easy to integrate into an existing wallet that already has its own key management, storage, blockchain download logic, etc. Written in rust, it can be compiled out to almost any platform (wasm, shared library, static library, etc) rather easily and can be called from virtually any language.
As of the past month or so, the API has started to stabilize and I'm starting to look at exploring possibly integrations. It's not at the "you should put money in this" point yet, but it's been more than capable of opening/closing channels/sending/receiving payments, etc for months and we're chipping away at error handling and test coverage. Happy to chat more with folks if there's interest.
[/shill]

**Comment by chiselinc** [comment]
Pinging on this thread, especially after stumbling across the well-articulated and well-earned shill from Blue Matt half a year ago.

I'm doing livechat support for a BTC Pay Server merchant with Lightning integration. Just walked a customer through installing a mobile wallet that *does* support Lightning, and potentially moving 

Personally Mycelium was my main mobile wallet from 2016 through early 2018, when I jumped ship after your failure to adopt SegWit (which you finally apparently did?). 

Just wanted to point out here, after the half hour of work I did with this awesome customer, that you'll continue to lose users if you fail to prioritize legit innovations from Bitcoin core development. This isn't just a matter of "Lightning would be nice"-- the next bull is coming soon, and fees will be untenable again if we don't start scaling to Layer 2.

But I'm talking to the company that ran an ICO for a Bitcoin wallet so maybe I'm wasting my breath regarding "priorities"

**Comment by woeisme** [comment]
Bumping this as well since it's now 2021 and Mycelium seems to be a bit behind the curve now. Support Lightning already Mycelium, there's tons of working wallet implementations, you have lots of loyal users out there that will eventually have to shift away without support. Hell you even had Matt Corallo offering to help, what more do you need?
